geo/src/main/java/org/odk/collect/geo/geopoint/GeoPointMapActivity.java

```diff
@@ -156,6 +157,26 @@ public void onCreate(Bundle savedInstanceState) {
 
         MapFragment mapFragment = ((FragmentContainerView) findViewById(R.id.map_container)).getFragment();
         mapFragment.init(this::initMap, this::finish);
+
+        // https://github.com/getodk/collect/issues/5293
```

Comment:
Instead of referencing an issue here, I'd prefer (as always) to document this with a test. This also ensures the behaviours survival! I'd be most tempted to add a test to `GeoPointMapActivityTest`. This is already set up to test the Activity, and you can use Espresso (there's `EspressoHelper` for basic operations as well) to interact with and assert on the UI.

Reply:
Sorry @seadowg but I can't actually see how to adapt `GeoPointMapActivityTest` as suggested. Can you point me to a comparable worked example?

Or is the idea of testing this feature somewhat of a counsel of perfection anyway?
Especially as the analogous feature in `GeoPolyActivity` isn't tested.

Reply:
> Sorry @seadowg but I can't actually see how to adapt GeoPointMapActivityTest as suggested. Can you point me to a comparable worked example?

I'd imagine tests that starts the Activity, selects a point and simulates a back press, and then checks that a dialog is shown and that cancel and ok behave as expected (probably two separate tests). It'd also be good to check other cases like when no point is selected or the Activity is started with a saved location and no changes are made.

> Or is the idea of testing this feature somewhat of a counsel of perfection anyway?
Especially as the analogous feature in GeoPolyActivity isn't tested.

We don't accept contributions without tests unless testing isn't possible or reasonable. AndroidX Test and Espresso tooling makes testing this very possible however. You're right that there is a lot of untested code in Collect, but that's something we're actively looking to improve upon so that we can be more confident about making changes in the future (more on that [here](https://github.com/getodk/collect/blob/master/docs/TEST-GUIDELINES.md) and [here](https://github.com/getodk/collect/blob/master/docs/STATE.md)). Adding more untested code is definitely not something that would help that effort!